pyproject.toml

```diff
@@ -0,0 +1,71 @@
+[build-system]
+requires = ["hatchling >= 1.26"]
+build-backend = "hatchling.build"
+
+# General metadata for the project
+[project]
+name = "pinns-for-comp-mech"
+version = "2025.0.0"
+license = { text = "MIT" }
+authors = [
+  {name = "Tarik Sahin", email = "tarik.sahin@unibw.de"},
+  {name = "Max von Danwitz", email = "max.danwitz@unibw.de"},
+  {name = "Daniel Wolff", email = "d.wolff@unibw.de"},
+]
+maintainers = [
+  {name = "Tarik Sahin", email = "tarik.sahin@unibw.de"},
+  {name = "Daniel Wolff", email = "d.wolff@unibw.de"},
+]
+description = "A library for PINNs on Computational Mechanics"
+keywords = ["physics-informed neural networks", "computational mechanics", "contact mechanics"]
+readme = "README.md"
+
+# Online references about the project
+[project.urls]
+Homepage = "https://github.com/imcs-compsim/pinns_for_comp_mech"
+Repository = "https://github.com/imcs-compsim/pinns_for_comp_mech.git"
+Issues = "https://github.com/imcs-compsim/pinns_for_comp_mech/issues"
+"Bug Tracker" = "https://github.com/imcs-compsim/pinns_for_comp_mech/issues?q=is%3Aissue%20state%3Aopen%20label%3Abug"
+
+# Dependencies of the project that need to be installed to use this project properly
+requires-python = ">= 3.8"
+dependencies = [
+  "deepxde",
+  "gmsh",
```

Comment:
Based on the conda installation, there were some errors for the gmsh installation. I can't remember if it was only on the cluster, so, I came up with the python-gmsh package as solution. One needs to check

Reply:
This dependencies are not installed via `conda` but via `pip`. It's literally the same as if they were contained in a `requirements.txt` file, just that the installation of the dependencies happens automatically. 

But you raise a good point, I will check whether the new setup also works on our cluster!

Reply:
[  - python-gmsh](https://github.com/imcs-compsim/pinns_for_comp_mech/blob/4fd992ac3a53f8bbae2d1ae48d0d0914a40849f9/environment.yml#L6)

Since the gmsh package fails on the cluster (if I correctly remember), I used python-gmsh for the cluster installation, which is done based on the conda. So it is different from the one in `requirements.txt`.

Reply:
Ahh okay, I see, thanks for clarifying, I'll look into that!

Reply:
Okay so, after careful testing on both my local machine and our cluster, I can confirm that indeed we no longer need `python-gmsh`. All the geometric functionality that relies on `gmsh` seems to be working just fine with the version that is installed via `pip`. 

This might be both due to upgrades on the `gmsh` side as well as the new Ubuntu system on our cluster. But this should not be an issue anymore :)